### PR TITLE
[velero] Add defaultResticPruneFrequency option to helm chart

### LINKED ISSUE #321: Make defaultResticPruneFrequency customizable via helm values
**Describe the problem/challenge you have**
Currently there is no way to customize defaultResticMaintenanceFrequency

**Describe the solution you'd like**

Add --default-restic-prune-frequency flag to /velero server at templates/deployment.yaml 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
Useful links:
https://github.com/vmware-tanzu/velero/pull/1864
https://github.com/vmware-tanzu/velero/issues/1541
https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/server/server.go#L215

**Environment:**

- helm version (use `helm version`): 
- helm chart version and app version (use `helm list -n <YOUR NAMESPACE>`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by marinoborges**
#### Special notes for your reviewer:
Closes #321
Tested locally with success

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the values.yaml or README.md
- [X] Title of the PR starts with chart name (e.g. `[velero]`)

### CODE REVIEW

File: charts/velero/templates/deployment.yaml [hunk 1]
```diff
@@ -99,6 +99,9 @@ spec:
             {{- if .defaultVolumesToRestic }}
             - --default-volumes-to-restic
             {{- end }}
+            {{- with .defaultResticMaintenanceFrequency }}
```
Comment (jenting) [inline charts/velero/templates/deployment.yaml:102]: Can we change the name to align with the flag? `defaultResticPruneFrequency`

**Review by jenting — CHANGES_REQUESTED** [review]
Missing signed-off-by in your commit message.

File: charts/velero/templates/deployment.yaml (hunk 1 above)
Comment (marinoborges) [reply charts/velero/templates/deployment.yaml:102]: Sure

**Comment by marinoborges** [comment]
I did put the `Signed-off-by: Marino Borges <39623860+marinoborges@users.noreply.github.com>` at the extended description for every commit message.. is anything missing @jenting ?

**Comment by jenting** [comment]
I think the GitHub checks your actual email account (should be the email of your GitHub account's mail)

**Review by jenting — APPROVED** [review]
LGTM, thank you.

**Review by sseago — APPROVED** [review]

**Pull request merged**